### DISCUSSION TITLE
`nth-child` selector does not work in ItemsControl  scope style setting

### DISCUSSION THREAD

**Discussion by liugangnhm** [AvaloniaUI/Avalonia#14238] [Q&A] [answered]
## Describe the bug
`nth-child` selector does not work in ItemsControl  scope style setting

## To Reproduce

```xaml
 <ItemsControl ItemsSource="{Binding Data}">
       
       <ItemsControl.Styles>
           <Style Selector="CheckBox:nth-child(n+2)">
               <Setter Property="Background" Value="Aqua"></Setter>
           </Style>
       </ItemsControl.Styles>
       
       <ItemsControl.ItemsPanel>
           <ItemsPanelTemplate>
               <StackPanel Orientation="Horizontal" Spacing="8"></StackPanel>
           </ItemsPanelTemplate>
       </ItemsControl.ItemsPanel>
       
       <ItemsControl.ItemTemplate>
           <DataTemplate>
               <CheckBox Content="{Binding}"></CheckBox>
           </DataTemplate>
       </ItemsControl.ItemTemplate>
   </ItemsControl>
```

## Screenshots
![image](https://github.com/AvaloniaUI/Avalonia/assets/743845/788a6a59-88e1-4074-87d0-9492c32e2f27)

## Environment 

- OS: windows11
- Avalonia-Version:11.0.6

## Additional context

when remove `nth-child`, the style will apply correctly

**Comment by maxkatz6** [comment, marked as answer]
One of the specifics of ItemsControl (and other ItemsControl delivered controls), that all children are wrapped in the virtualized/reused container. Usually of ContentPresenter type.

Essentially visual tree looks like this:
```
ItemsControl
- ContentPresenter 1
- - CheckBox 1
- ContentPresenter 2
- - CheckBox 2
- ...
```

You can see it in DevTools as well.

Correct selector would be `ContentPresenter:nth-child(n+2) CheckBox`

**Reply by liugangnhm** [reply]
thanks very much , it works correctly
